src/toil_scripts/tools/preprocessing.py

```diff
@@ -82,7 +85,34 @@ def run_samtools_index(job, bam_id):
     return job.fileStore.writeGlobalFile(os.path.join(work_dir, 'sample.bam.bai'))
 
 
-def run_picard_create_sequence_dictionary(job, ref_id):
+def samtools_view(job, bam_id, flag='0', mock=False):
+    """
+    Filters BAM file using SAM bitwise flag
+
+    :param bam_id str: BAM FileStoreID
+    :param flag str: SAM bitwise flags
+    :param mock bool: If True, run in mock mode
+    :return str: BAM fileStoreID
+    """
+    work_dir = job.fileStore.getLocalTempDir()
+    job.fileStore.readGlobalFile(bam_id, os.path.join(work_dir, 'input.bam'))
+    outputs = {'output.bam': None}
+    cores = multiprocessing.cpu_count()
```

Comment:
As an aside, someone might want to run this job on fewer cores than the machine has (e.g., `job.wrapJobFn(..., cores = 2)`), so we should probably avoid autodetecting the core count.